### ISSUE TITLE
Music Sync fails on extra large libraries

### ISSUE THREAD

**Issue by mcarlton00** [jellyfin/jellyfin-kodi#386]
Hi, I'm experiencing a similar issue:
I tried to sync the music-library and after 12:20 min. I received an error-message (Something went wrong).
In this library are about 45'000 songs.
If any log-files would help you for this topic, I would be glad to share them.

_Originally posted by @n-buck in https://github.com/jellyfin/jellyfin-kodi/issues/380#issuecomment-688072472_

Logs: https://paste.kodi.tv/ewodijidiv

**Comment by mcarlton00** [comment]
Moving this to a new issue since I don't think this was the cause of #380 

Confirms my suspicions that the music sync revamp will fail on extra large libraries due to the number of items returned by the server in a single api call.  The server is too slow to gather and return all the items in a reasonable amount of time, and I would imagine even if that worked then lower end devices (such as Pis or older PCs) might OOM themselves trying to process that much json in a single pass.

TLDR: Music syncing needs another revamp

**Comment by niawag** [comment]
Hi and sorry to barge in, I'm having the same problem with syncing a large music collection (~30k songs): stuck at 0% and "Something went wrong" after 10-15 minutes. I can see that you are working on this and would be glad to help testing this.
Can I do somethnig? If I want to test your pull (#387), can I just replace the concerned file in the addon folder or is there another way to do this?

**Comment by mcarlton00** [comment]
No need to apologize, I need all the guinea pigs I can find.  My library is only 9k songs and doesn't time out, which is why I didn't realize the issue when I reworked this the first time.

You can change the files directly, however the easier method is probably to download the linked zip for your Kodi version from [this](https://github.com/jellyfin/jellyfin-kodi/pull/387#issuecomment-688590931) comment and install that directly (Settings -> Add-Ons -> Install from zip file)

**Comment by niawag** [comment]
Great, I should be able to test it @home tonight, I'll report the result ASAP

But keep in mind I'm no guinea pig, I'm a guinea bear! ;)
